app/controllers/legacy/authentications_controller.rb

```diff
@@ -0,0 +1,30 @@
+module Legacy
+  class AuthenticationsController < ApplicationController
+    before_action :reauthenticate_user_if_signin_is_too_old!
+
+    # Remove authentication method (OAuth provider) from account
+    def destroy
+      handle_with(
+        AuthenticationsDelete,
+        success: lambda do
+          authentication = @handler_result.outputs.authentication
+          security_log :authentication_deleted,
+                      authentication_id: authentication.id,
+                      authentication_provider: authentication.provider,
+                      authentication_uid: authentication.uid
+          render status: :ok,
+                plain: (I18n.t :"controllers.authentications.authentication_removed",
+                              authentication: params[:provider].titleize)
+        end,
+        failure: lambda do
+          render status: 422, plain: @handler_result.errors.map(&:message).to_sentence
```

Comment:
consider moving this demeteried train of dots into one method:    @handler_result.human_friendly(), so that it can be used anywhere and unit tested inside the handler_result code.

Reply:
This is legacy code that is expected to be deleted soon. So honestly I don't think it's worth refactoring legacy code ATM. I admit that I did reuse that legacy code though, in SocialAuthController#remove_auth_strategy and I was tempted to change that line to `@handler_result.errors.first&.message` but it still suffers from the same problem anyway, so I'm just going to leave it untouched for now—again, because it's legacy code that we intend to get rid of replace soon.